packages/webpack/src/config/base.js

```diff
@@ -43,17 +36,29 @@ export default class WebpackBaseConfig {
 
   get nuxtEnv() {
     return {
-      isDev: this.options.dev,
+      isDev: this.dev,
       isServer: this.isServer,
       isClient: !this.isServer,
       isModern: !!this.isModern
     }
   }
 
+  get mode() {
+    return this.dev ? 'development' : 'production'
+  }
+
+  get dev() {
+    return this.context.options.dev
+  }
+
+  get buildOpts() {
```

Comment:
Might be just personal taste but I'd prefer writing out `options` 🙈 

```suggestion
  get buildOptions() {
```

Reply:
OK